server/services/prisonerSearchService.ts

```diff
@@ -1,16 +1,64 @@
 import type { PrisonerSummary } from 'viewModels'
+import type { Prisoner } from 'prisonerSearchApiClient'
 import { HmppsAuthClient, PrisonerSearchClient } from '../data'
 import toPrisonerSummary from '../data/mappers/prisonerSummaryMapper'
+import logger from '../../logger'
+import PrisonerSearchStore from '../data/prisonerSearchStore/prisonerSearchStore'
+
+const PRISONER_CACHE_TTL_DAYS = 1
 
 export default class PrisonerSearchService {
   constructor(
     private readonly hmppsAuthClient: HmppsAuthClient,
     private readonly prisonerSearchClient: PrisonerSearchClient,
+    private readonly prisonerSearchStore: PrisonerSearchStore,
   ) {}
 
   async getPrisonerByPrisonNumber(prisonNumber: string, username: string): Promise<PrisonerSummary> {
-    const systemToken = await this.hmppsAuthClient.getSystemClientToken(username)
-    const prisoner = await this.prisonerSearchClient.getPrisonerByPrisonNumber(prisonNumber, systemToken)
+    let prisoner: Prisoner
```

Comment:
The service method to get a prisoner by prison number now looks to return it from redis first
It the prisoner is in redis (the store class) then use it
If the prisoner is not in redis then retrieve if from the API (the previous behaviour), and store it in redis